guides/pages/how-to-synchronize-your-yjs-storage-data-to-a-planetscale-mysql-database.mdx

```diff
@@ -0,0 +1,584 @@
+---
+meta:
+  title:
+    "How to synchronize your Yjs Storage data to a PlanetScale MySQL database"
+  description:
+    "Learn how to automatically update your PlanetScale MySQL database with
+    changes from your Yjs application."
+---
+
+Liveblocks allows you to build collaborative applications with
+[Yjs](https://yjs.dev/), which is tightly integrated into our infrastructure.
+Using our Webhooks and REST API, you can then retrieve the Yjs document data as
+it changes, and store it in your database.
+
+## What we’re building
+
+In this guide we’ll be linking a Yjs application up to a
+[PlanetScale](https://planetscale.com/) MySQL database so that Yjs document data
+is automatically synchronized. This is enabled through the following:
+
+- [YDocUpdated](/docs/platform/webhooks#YDocUpdatedEvent) Webhook event
+- [Get Yjs Document](/docs/api-reference/rest-api-endpoints#get-rooms-roomId-ydoc)
+  REST API
+
+<Banner title="Have a project ready">
+
+This guide assumes you already have a Liveblocks Yjs project set up. If haven’t
+already got one, you can open our getting started page, select a
+[text editor](/docs/get-started/text-editor) or
+[code editor](/docs/get-started/code-editor), both of which use Yjs, and come
+back after you’re set up.
+
+</Banner>
+
+### This specific webhook is throttled
+
+Note that the [YDocUpdated](/docs/platform/webhooks#YDocUpdatedEvent) webhook
+event is throttled at a rate of _once every 5 minutes_. This is because Yjs can
+update up to _60 times per second_, and it would be impractical to run the
+webhook this frequently.
+
+## Create an endpoint in your project
+
+When a webhook event is triggered, it can send POST request to the back end in
+your project, and from within there we can update the database. In this guide,
+we’ll be using a Next.js route handler (API endpoint) as an example, but other
+frameworks work similarly.
+
+In order to use webhooks, we’ll need to retrieve the `headers` and `body` from
+the request. Here’s the basic endpoint we’ll be starting from:
+
+```ts
+export async function POST(request: Request) {
+  const body = await request.json();
+  const headers = request.headers;
+
+  // Handle webhooks and database
+  // ...
+
+  return new Response(null, { status: 200 });
+}
+```
+
+Create this endpoint in your project, and make it available on `localhost` at
+the following URL:
+
+```
+/api/liveblocks-database-sync
+```
+
+### Testing webhooks locally
+
+Running webhooks locally can be difficult, but one way to do this is to use a
```

Comment:
I got stuck at that point last time, this would have been helpful. Glad we're going to have a guide that explains how to test things locally.